File: engine/engine-model/src/main/scala/tools/packet/wedding/WeddingInvitation.scala
package tools.packet.wedding

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class WeddingInvitation(private var _groom: String, private var _bride: String) extends PacketInput {
  def groom: String = _groom

  def bride: String = _bride

  override def opcode(): SendOpcode = SendOpcode.MARRIAGE_RESULT
}

File: engine/engine-model/src/main/java/tools/packet/wedding/WeddingInvitation.java
package tools.packet.wedding;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record WeddingInvitation(String groom, String bride) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MARRIAGE_RESULT;
   }
}